gnovm/cmd/gno/util.go

```diff
@@ -25,36 +21,6 @@ func isFileExist(path string) bool {
 	return err == nil
 }
 
-func gnoFilesFromArgs(args []string) ([]string, error) {
-	paths := []string{}
-	for _, arg := range args {
-		info, err := os.Stat(arg)
-		if err != nil {
-			return nil, fmt.Errorf("invalid file or package path: %w", err)
-		}
-		if !info.IsDir() {
-			curpath := arg
-			paths = append(paths, curpath)
-		} else {
-			err = filepath.WalkDir(arg, func(curpath string, f fs.DirEntry, err error) error {
-				if err != nil {
-					return fmt.Errorf("%s: walk dir: %w", arg, err)
-				}
-
-				if !isGnoFile(f) {
-					return nil // skip
-				}
-				paths = append(paths, curpath)
-				return nil
-			})
-			if err != nil {
-				return nil, err
-			}
-		}
-	}
-	return paths, nil
-}
-
 func gnoPackagesFromArgs(args []string) ([]string, error) {
```

Comment:
I think we can just completely eliminate this spaghetti method implementation
```go
func gnoPackagesFromArgs(args []string) ([]string, error) {
	paths := make([]string, 0)

	for _, argPath := range args {
		info, err := os.Stat(argPath)
		if err != nil {
			return nil, fmt.Errorf("invalid file or package path: %w", err)
		}

		if !info.IsDir() {
			paths = append(paths, ensurePathPrefix(argPath))

			continue
		}

		// Gather package paths from the directory
		err = walkDirForGnoFiles(argPath, func(path string) {
			paths = append(paths, ensurePathPrefix(path))
		})
		if err != nil {
			return nil, fmt.Errorf("unable to walk dir: %w", err)
		}
	}

	return paths, nil
}

func ensurePathPrefix(path string) string {
	if filepath.IsAbs(path) {
		return path
	}

	// cannot use path.Join or filepath.Join, because we need
	// to ensure that ./ is the prefix to pass to go build.
	// if not absolute.
	return "." + string(filepath.Separator) + path
}

func walkDirForGnoFiles(root string, addPath func(pkg string)) error {
	visited := make(map[string]struct{})

	walkFn := func(currPath string, f fs.DirEntry, err error) error {
		if err != nil {
			return fmt.Errorf("%s: walk dir: %w", root, err)
		}

		if f.IsDir() || !isGnoFile(f) {
			return nil
		}

		parentDir := filepath.Dir(currPath)
		if _, found := visited[parentDir]; found {
			return nil
		}

		visited[parentDir] = struct{}{}

		addPath(parentDir)

		return nil
	}

	return filepath.WalkDir(root, walkFn)
}
```

Reply:
applied!